### PR TITLE
XFEMM docs and tests for each problem

### PR DESCRIPTION
**Pull request by Anvar-Khamitov**
This pull request is ready for review.

### CODE REVIEW

**Review by npetersen2 — COMMENTED** [review]
Might want to consider adding more test cases. For example, consider arcs and segments in different quadrants to stress more `if` statements in code, i.e., make some things negative. This can tend to be problematic. Also, consider cases when arcs and lines overlap at "corner cases," like 180 degrees, etc.

**Comment by elsevers** [comment]
This PR is to implement issue #80.

**Comment by elsevers** [comment]
This PR also implements issue #87

**Comment by Anvar-Khamitov** [comment]
This PR is ready to be merged. It solves the issue #87 . To complete the issue #80 , another PR will be created if this PR is merged.

File: tools/xfemm/XFEMM.m [hunk 1]
```diff
@@ -437,9 +443,10 @@ function plot(obj)
             
             % Remove partially overlapping arcs and make reconnections
             obj.FemmProblem = removePartiallyOverlappingArcSegments(obj);
-            
+
```
Comment (npetersen2) [inline tools/xfemm/XFEMM.m]: Remove this whitespace

File: tools/xfemm/XFEMM.m [hunk 2]
```diff
@@ -437,9 +443,10 @@ function plot(obj)
             
             % Remove partially overlapping arcs and make reconnections
             obj.FemmProblem = removePartiallyOverlappingArcSegments(obj);
-            
+                                   
             % Return modified FemmProblem struct
             FemmProblem = obj.FemmProblem;
+
```
Comment (npetersen2) [inline tools/xfemm/XFEMM.m]: Remove this whitespace

**Review by npetersen2 — CHANGES_REQUESTED** [review]
Please remove extra whitespace changes.

**Review by elsevers — APPROVED** [review]
Looks good

**Review by npetersen2 — APPROVED** [review]
Looks good to me.

**Pull request merged**